infrastructure/provider.tf

```diff
@@ -0,0 +1,3 @@
+provider "azurerm" {
```

Comment:
I think you backend block defined as well  `backend "azurerm" {}`

full example:
https://github.com/hmcts/github-management/blob/main/terraform/providers.tf#L3